keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -386,4 +386,16 @@ class SqlVerificationRepository(
           ?.asTable(alias, ind, environmentName, artifactReference, artifactVersion)
   }
 
+  /**
+   * Set a share mode lock on a select query to prevent phantom reads in a transaction.
+   *
+   * In MySQL 5.7, this is `LOCK IN SHARE MODE`
+   * See https://dev.mysql.com/doc/refman/5.7/en/innodb-locking-reads.html
+   */
+  private fun <R : Record?> SelectForUpdateStep<R>.lockInShareMode(): SelectOptionStep<R> =
+    if(useLockingRead) {
+      this.forShare()
+    } else {
+      this
+    }
```

Comment:
This is going to lock across the cluster? What's the scope of that lock (query / table?) What happens if the instance holding the lock terminates unexpectedly?

Reply:
I communicated with @robfletcher about this out of band. This is a read-write lock, meaning that it won't block reads. The lock is scoped to the transaction, so if an instance terminates during the transaction then mysql will release the locks once it detects the connection has been torn down.